system/HTTP/IncomingRequest.php

```diff
@@ -198,6 +196,7 @@ public function detectLocale($config)
     protected function detectURI(string $protocol, string $baseURL)
     {
         // Passing the config is unnecessary but left for legacy purposes
+        // @TODO For what? Why clone?
```

Comment:
The clone prevents URI testing from changing the App Config globally.

Reply:
The reason for `clone` is only for testing, I would like to remove it.

Reply:
And deprecated the baseURL input? Is that really only for testing?

Reply:
Both `$protocol` and `$baseURL` of `detectURI()` are not needed.

__construct():
https://github.com/codeigniter4/CodeIgniter4/blob/11680d6b1c974a72afa2d083a2ceb8954866ed20/system/HTTP/IncomingRequest.php#L172

https://github.com/codeigniter4/CodeIgniter4/blob/11680d6b1c974a72afa2d083a2ceb8954866ed20/system/HTTP/IncomingRequest.php#L198-L205